static/app/utils/useDismissAlert.tsx

```diff
@@ -0,0 +1,59 @@
+import {useCallback, useEffect, useState} from 'react';
+
+import localStorage from 'sentry/utils/localStorage';
+
+type Opts = {
+  /**
+   * Key in localstorage.
+   * Use a format like: `${organization.id}:my-feature`
+   */
+  key: string;
+
+  /**
+   * Number of days before the dismissal is expired.
+   * After expiration the the user will need to re-dismiss things.
+   */
+  expirationDays?: number;
+};
+
+const MS_PER_DAY = 1000 * 60 * 60 * 24;
+
+/**
+ * Nominally for tracking dismissal/acknowledgement of in-app alerts and
+ * notifications, specfically those that don't need to be persisted server side.
+ *
+ * Dismissal can be 'permanent' or have an expiration after some number of days.
+ *
+ * The real lifecycle depends on whether users reset localStorage or not, if you
+ * need something really permanent then save the users' preference to the server.
+ */
+function useDismissAlert({expirationDays = Number.MAX_SAFE_INTEGER, key}: Opts) {
+  const [isDismissed, setDismissed] = useState(false);
+
+  useEffect(() => {
+    const val = localStorage.getItem(key);
```

Comment:
Would it make sense to use the `useLocalStorageState` hook?

Reply:
yeah, that's a handy hook. 
It's frustrating to test that something was actually saved to localStorage though, I don't see any tests that wire things up to assert that kind of thing either. Something about the queueMicrotask call, the `jest.run*` methods don't seem to help either :(

Reply:
you could rely on it's tests right and not need to test that it's actually putting stuff in local storage, instead you can spy on it's functions and check that those are called maybe?